float32 xPosition
float32 zPosition
float32 xVelocity
float32 zVelocity
float32 time